engine/src/p2p/core.rs

```diff
@@ -539,13 +541,17 @@ impl P2PContext {
 			return
 		}
 
+		let mut previous_activity = None;
```

Comment:
Could we set this to `tokio::time::Instant::now` here instead of using an option in `connect_to_peer`?